diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index de5222645..eb27e8b4e 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -448,6 +448,8 @@ foreach(arch ${archs})
                 use_udev=true
                 is_cfi=false
                 use_ozone=true
+                media_use_openh264=true
+                rtc_use_h264=true
                 ozone_auto_platforms=false
                 ozone_platform_headless=false
                 ozone_platform_external=true
